src/main/java/core/basesyntax/MainApp.java

```diff
@@ -11,6 +11,56 @@
  * - В классе Main создать добавить по одному екземпляру каждого класса.
  * - Отправить все машини на работу и остановить работу всех машин.
  */
-public class MainApp {
+abstract class Machine {
+    public abstract void doWork();
+    public abstract void stopWork();
+}
+class Track extends Machine{
+
+    @Override
+    public void doWork() {
+        System.out.println("The track started working");
+    }
+
+    @Override
+    public void stopWork() {
+        System.out.println("The track stopped working");
+    }
+}
+class Bulldozer extends Machine{
+
+    @Override
+    public void doWork() {
+        System.out.println("The bulldozer started working");
+    }
 
+    @Override
+    public void stopWork() {
+        System.out.println("The bulldozer stopped working");
+    }
+}
+class Excavator extends Machine{
+
+    @Override
+    public void doWork() {
+        System.out.println("The excavator started working");
+    }
+
+    @Override
+    public void stopWork() {
+        System.out.println("The excavator stopped working");
+    }
+}
+public class MainApp {
+    public static void main(String[] args) {
+        Track track = new Track();
```

Comment:
be attentive with [checklist](https://mate-academy.github.io/jv-program-fulltime/01_core/checklist/hw_02.html)
- Use abstract class to create Track, Bulldozer and Excavator instances.

Example:
Animal cat = new Cat();